hamster_gtk/preferences/preferences_dialog.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+
+
+# This file is part of 'hamster-gtk'.
+#
+# 'hamster-gtk' is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# 'hamster-gtk' is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with 'hamster-gtk'.  If not, see <http://www.gnu.org/licenses/>.
+
+
+"""This module provides the preferences dialog."""
+
+
+from __future__ import absolute_import, unicode_literals
+
+import collections
+from gettext import gettext as _
+
+import hamster_lib
+from gi.repository import GObject, Gtk
+
+from hamster_gtk.preferences.widgets import (ConfigComboBoxText, ConfigWidget,
+                                             DurationEntry,
+                                             EditableFileChooser, TimeEntry)
+
+
+class PreferencesDialog(Gtk.Dialog):
+    """A dialog that shows and allows editing of config settings."""
+
+    def __init__(self, parent, app, initial, *args, **kwargs):
+        """
+        Instantiate dialog.
+
+        Args:
+            parent (Gtk.Window): Dialog parent.
+            app (HamsterGTK): Main app instance. Needed in order to retrieve
+                and manipulate config values.
+        """
+        super(PreferencesDialog, self).__init__(*args, **kwargs)
+
+        self._parent = parent
+        self._app = app
+
+        self.set_transient_for(self._parent)
+
+        db_engines = [('sqlite', _('SQLite')), ('postgresql', _('PostgreSQL')),
+            ('mysql', _('MySQL')), ('oracle', _('Oracle')), ('mssql', _('MSSQL'))]
```

Comment:
I am not really convinced about this, for example Japanese might want to transliterate the names to Katakana, Russians might want to use Cyrillic or Thai whatever their script is called – [มายเอสคิวเอล](https://th.wikipedia.org/wiki/%E0%B8%A1%E0%B8%B2%E0%B8%A2%E0%B9%80%E0%B8%AD%E0%B8%AA%E0%B8%84%E0%B8%B4%E0%B8%A7%E0%B9%80%E0%B8%AD%E0%B8%A5). Yay for different cultures.